Mathlib/Analysis/MellinTransform.lean

```diff
@@ -123,19 +121,23 @@ theorem mellin_div_const (f : ℝ → ℂ) (s a : ℂ) : mellin (fun t => f t /
   simp_rw [mellin, smul_eq_mul, ← mul_div_assoc, integral_div]
 #align mellin_div_const mellin_div_const
 
-theorem mellin_comp_rpow (f : ℝ → E) (s : ℂ) {a : ℝ} (ha : a ≠ 0) :
+theorem mellin_comp_rpow (f : ℝ → E) (s : ℂ) (a : ℝ) :
     mellin (fun t => f (t ^ a)) s = |a|⁻¹ • mellin f (s / a) := by
-  -- note: this is also true for a = 0 (both sides are zero), but this is mathematically
-  -- uninteresting and rather time-consuming to check
-  simp_rw [mellin]
-  conv_rhs => rw [← integral_comp_rpow_Ioi _ ha, ← integral_smul]
-  refine' set_integral_congr measurableSet_Ioi fun t ht => _
-  dsimp only
-  rw [← mul_smul, ← mul_assoc, inv_mul_cancel (mt abs_eq_zero.1 ha), one_mul, ← smul_assoc,
-    real_smul]
-  rw [ofReal_cpow (le_of_lt ht), ← cpow_mul_ofReal_nonneg (le_of_lt ht), ←
-    cpow_add _ _ (ofReal_ne_zero.mpr <| ne_of_gt ht), ofReal_sub, ofReal_one, mul_sub,
-    mul_div_cancel' _ (ofReal_ne_zero.mpr ha), add_comm, ← add_sub_assoc, mul_one, sub_add_cancel]
+  /- This is true for `a = 0` as all sides are undefined but turn out to vanish thanks to our
+  convention. The interesting case is `a ≠ 0` -/
+  rcases eq_or_ne a 0 with rfl|ha
+  · by_cases hE : CompleteSpace E
+    · simp [integral_smul_const, mellin, setIntegral_Ioi_zero_cpow]
+    · simp [integral, mellin, hE]
+  · simp_rw [mellin]
+    conv_rhs => rw [← integral_comp_rpow_Ioi _ ha, ← integral_smul]
+    refine' set_integral_congr measurableSet_Ioi fun t ht => _
+    dsimp only
+    rw [← mul_smul, ← mul_assoc, inv_mul_cancel (mt abs_eq_zero.1 ha), one_mul, ← smul_assoc,
+      real_smul]
+    rw [ofReal_cpow (le_of_lt ht), ← cpow_mul_ofReal_nonneg (le_of_lt ht), ←
+      cpow_add _ _ (ofReal_ne_zero.mpr <| ne_of_gt ht), ofReal_sub, ofReal_one, mul_sub,
+      mul_div_cancel' _ (ofReal_ne_zero.mpr ha), add_comm, ← add_sub_assoc, mul_one, sub_add_cancel]
```

Comment:
Since this is the interesting case, I would rather not indent it
```suggestion
  simp_rw [mellin]
  conv_rhs => rw [← integral_comp_rpow_Ioi _ ha, ← integral_smul]
  refine' set_integral_congr measurableSet_Ioi fun t ht => _
  dsimp only
  rw [← mul_smul, ← mul_assoc, inv_mul_cancel (mt abs_eq_zero.1 ha), one_mul, ← smul_assoc,
    real_smul]
  rw [ofReal_cpow (le_of_lt ht), ← cpow_mul_ofReal_nonneg (le_of_lt ht), ←
    cpow_add _ _ (ofReal_ne_zero.mpr <| ne_of_gt ht), ofReal_sub, ofReal_one, mul_sub,
    mul_div_cancel' _ (ofReal_ne_zero.mpr ha), add_comm, ← add_sub_assoc, mul_one, sub_add_cancel]
```